src/sh/python-lint/run-in-poetry-subdir.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+# cspell:words realpath, autoflake, venv
+
+# Capture the first argument, which is the command we're wrapping for
+# the pre-commit hook.
+COMMAND=$1
+# Then skip it so we can pass "$@" to the command.
+shift
+
+# Capture and skip like above.
+ERROR_MESSAGE=$1
+shift
+
+# This is the path passed in to the script from the pre-commit hook,
+# it tells poetry where to look for the files.
+POETRY_SUBDIRECTORY="./src/python/hooks"
+
+RELATIVE_PATHS=""
+
+# Convert all paths to relative paths.
+for path in "$@"; do
+	RELATIVE=$(realpath --relative-to="$POETRY_SUBDIRECTORY" "$path")
+	RELATIVE_PATHS="$RELATIVE_PATHS $RELATIVE"
+done
+
+cd $POETRY_SUBDIRECTORY || exit 1
+
+# Then run the script passed into this script, with the relative paths.
+# This is so we can define individual pre-commit hooks for each linter,
+# each with their own output status codes.
+fail=false
+
+eval $COMMAND $RELATIVE_PATHS || fail=true
```

Comment:
Most of the pre-commit hooks call on this without specifying `poetry install` a priori

Should the `poetry-check`.sh run `poetry install`?

Reply:
Will add to list of things to discuss, but just to clarify, specifically in CI the poetry executable and the poetry environment are both installed in the `pre-commit.yaml` workflow, and `poetry-check` is primarily to detect out-of-sync `poetry.lock` files.

Installing the poetry executable and poetry env is a different flow for users, so I didn't include it in `poetry-check.sh`, because I wasn't sure where we'd expect users to do it (or if we'd do it for them in a script)

Reply:
Marking resolved, captured in https://github.com/econia-labs/econia-v5/pull/42#discussion_r1544847872